### PR TITLE
Move prow/{plugins,config}.yaml to config/

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR moves the `prow/plugins.yaml` and `prow/config.yaml` file to the `config` dir in an attempt to separate config of the prow.k8s.io instance from prow code. This way, reviewers for the Prow code do not get pinged for config changes of prow.k8s.io.

This PR is basically string replace of all occurences of `prow/{plugins,config}.yaml`  with `config/{plugins,config}.yaml`, then manually git adding the parts that seem appropriate.

Open question: Are the OWNERS for `config/` the right ones for this?

### CODE REVIEW

File: config/BUILD.bazel [hunk 1]
```diff
@@ -4,12 +4,12 @@ sh_binary(
     name = "mkpj",
     srcs = ["mkpj_binary"],
     args = [
-        "--config-path=$(location //prow:config.yaml)",
+        "--config-path=$(location //config:config.yaml)",
```
Comment (fejta) [inline config/BUILD.bazel]: Can we put this in config/prow/*.yaml?

File: config/plugins.yaml [hunk 2]
```diff
@@ -345,9 +345,9 @@ config_updater:
     label_sync/labels.yaml:
       name: label-config
       namespace: test-pods
-    prow/config.yaml:
+    config/config.yaml:
```
Comment (fejta) [inline config/plugins.yaml]: I suspect we'll want the file to exist in both locations for at least one commit.

File: prow/cmd/deck/github_oauth_setup.md [hunk 3]
```diff
@@ -112,11 +112,11 @@ When testing locally, pass the path to your secrets to `deck` using the `--githu
 
 Run the command:
 
-`go build . && ./deck --config-path=../../config.yaml --github-oauth-config-file=<PATH_TO_YOUR_GITHUB_OAUTH_SECRET> --cookie-secret=<PATH_TO_YOUR_COOKIE_SECRET> --oauth-url=/pr`
+`go build . && ./deck --config-path=../../../config/config.yaml --github-oauth-config-file=<PATH_TO_YOUR_GITHUB_OAUTH_SECRET> --cookie-secret=<PATH_TO_YOUR_COOKIE_SECRET> --oauth-url=/pr`
 
 Or, if you'd like to use bazel, run the command: 
 
-`bazel run //prow/cmd/deck -- --config-path=/absoluate/path/to/config.yaml --github-oauth-config-file=<PATH_TO_YOUR_GITHUB_OAUTH_SECRET> --cookie-secret=<PATH_TO_YOUR_COOKIE_SECRET> --oauth-url=/pr`
+`bazel run //prow/cmd/deck -- --config-path=/absoluate/pa../../../config/config.yaml --github-oauth-config-file=<PATH_TO_YOUR_GITHUB_OAUTH_SECRET> --cookie-secret=<PATH_TO_YOUR_COOKIE_SECRET> --oauth-url=/pr`
```
Comment (fejta) [inline prow/cmd/deck/github_oauth_setup.md]: This line shouldn't change

**Comment by alvaroaleman** [comment]
> Does this cause any rule visibility issues? Wouldn't be surprised to find out that is the case.

No idea, how would I find that out?

**Comment by fejta** [comment]
I didn't look through why bazel is failing. It will let you know if this is a problem. If rule A depends on B and B doesn't make itself visible to A then bazel will refuse to build A, see:



https://github.com/kubernetes/test-infra/blob/9b67b24c7b142f41728776c66e4a0e1af23fe315/config/BUILD.bazel#L1

https://github.com/kubernetes/test-infra/blob/dbbeb4216756c3e2bdffa7da6ac0bd97ead001e4/testgrid/util/gcs/BUILD.bazel#L10-L17

https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes

**Comment by alvaroaleman** [comment]
@fejta I've moved the configs to config/prow. Tests passed in 2c3fce26e0337e321c2619524da8079c2168dc32 when they weren't in prow/ anymore. I've re-added them to prow/ in 2c3fce26e0337e321c2619524da8079c2168dc32 so the next deployment still works :)

**Comment by Katharine** [comment]
I very inclined to punt this until after the release of 1.16, which should be tomorrow.

On the topic of the actual change: this looks like it changes some defaults - is that safe downstream? the defaults are admittedly very bizarre either way.

**Comment by alvaroaleman** [comment]
> On the topic of the actual change: this looks like it changes some defaults - is that safe downstream? the defaults are admittedly very bizarre either way.

Yeah, there is the change to `prow/plugins/config.go` which changes its default from `prow/{config,plugins}.yaml`. If someone relies on that, it will break.

The problem here is that while its easyish to find all references to a the old paths, its not as easy to find all places where the default of using the old path is being used, so I was afraid this will break prow.k8s.io if kept.

**Comment by chases2** [comment]
This change makes a copy of `config.yaml` and points the references to the new copy, but doesn't actually delete the old one, right? Or am I reading this diff wrong?

/kind oncall-hotlist
/hold cancel

I'm fine with this merging now and bumping tomorrow

**Comment by alvaroaleman** [comment]
> This change makes a copy of config.yaml and points the references to the new copy, but doesn't actually delete the old one, right? Or am I reading this diff wrong?

Thats accurate, but if one of prow/config.yaml or plugins.yaml changes on master, that won't get reflected by the copy. Git is smart enough to move the changes over to config/prow/{config,plugins}.yaml but not to do the same for the copy, as from its POV those are entirely new files

**Pull request merged**